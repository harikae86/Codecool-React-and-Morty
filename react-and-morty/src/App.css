@import url("https://fonts.googleapis.com/css2?family=Bangers&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,200;0,600;1,100;1,900&display=swap");

* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    text-decoration: none;
}

body,
html {
    font-family: "Montserrat", sans-serif;
    display: flex;
    flex-direction: column;
}

#root {
    height: 100%;
}
