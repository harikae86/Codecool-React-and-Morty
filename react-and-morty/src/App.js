import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import "./App.css";
import Home from "./components/Home";
import Characters from "./components/Characters/Characters";
import Locations from "./components/locations/Locations";
import CharacterDataDisplay from "./components/CharacterDisplay/CharacterDataDisplay";
import LocationDataDisplay from "./components/LocationDisplay/LocationDataDisplay";

function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/characters" element={<Characters />} />
        <Route path="/locations" element={<Locations />} />
        <Route path="/character/:id" element={<CharacterDataDisplay />} />
        <Route path="/location/:id" element={<LocationDataDisplay />} />
      </Routes>
    </Router>
  );
}

export default App;
