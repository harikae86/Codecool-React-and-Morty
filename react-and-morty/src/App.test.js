import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import { MemoryRouter, Link } from "react-router-dom";
import { createMemoryHistory } from "history";

test("There are test-cases that check whether the page loads successfuly", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

test("there is a test that tests whether there is a button for Characters", () => {
  render(<App />);
  const btn = screen.getByText("Characters");
  expect(btn).toBeVisible();
});

test("there is a test that tests whether there is a button for Locations", () => {
  render(<App />);
  const btn = screen.getByText("Locations");
  expect(btn).toBeVisible();
});

// test("routes to a new route", () => {
//   const history = createMemoryHistory();
//   history.push = jest.fn();

//   const { getByText } = render(
//     <MemoryRouter history={history}>
//       <Link to="/characters">Characters</Link>
//     </MemoryRouter>
//   );

//   fireEvent.click(getByText("Characters"));
//   expect(history.push).toHaveBeenCalledWith("/characters");
// });
