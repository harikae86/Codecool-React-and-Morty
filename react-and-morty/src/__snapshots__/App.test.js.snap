// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`There are test-cases that check whether the page loads successfuly 1`] = `
<div>
  <div>
    <img
      class="sc-bdvvtL gnkjgn"
      src="background.jpg"
    />
    <img
      class="sc-dkPtRN iemXGu"
      src="logo.png"
    />
    <div
      class="sc-gsDKAQ gWcezE"
    >
      <a
        class="HomeLink"
        href="/characters"
      >
        Characters
      </a>
      <a
        class="HomeLink"
        href="/locations"
      >
         
        Locations
      </a>
    </div>
    <div
      class="sc-hKwDye jOobhl"
    >
      <p
        class="Desc"
      >
        The Rick and Morty Wiki is an unofficial collaborative database for the hit [adult swim] series Rick and Morty, a science-fiction, black comedy series about genius inventor, Rick Sanchez, and his naive grandson, Morty Smith. The American animated television show is created by Justin Roiland and Dan Harmon. It premiered on Cartoon Network's [adult swim] block on December 2, 2013.
      </p>
    </div>
  </div>
</div>
`;
